deltacat/utils/profiling.py

```diff
@@ -0,0 +1,39 @@
+import ray
+from dataclasses import dataclass
+
+
+@dataclass
+class ClusterResourcesUsage:
```

Comment:
```suggestion
class ClusterResources:
```

Maybe this can be more concisely stated as `ClusterUtilization` here?

Also, if we change the API to take dictionaries of `cluster_resources` and `available_resources` from Ray, then this class becomes both more complete and maintainable IMO if we just make a 3rd dictionary of `used_resources` which takes the difference of all values with the same key between `cluster_resources` and `available_resources`. The values held in these dictionaries are already well-documented, so I think adding individual members for each of them isn't totally necessary, but creating dataclass members for commonly used values like you've done here could still at least help with IDE autocompletion.